overlord/configstate/configcore/homedirs.go

```diff
@@ -0,0 +1,178 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+	"github.com/snapcore/snapd/sandbox/apparmor"
+	"github.com/snapcore/snapd/sysconfig"
+)
+
+// For mocking in tests
+var (
+	osOpenFile = os.OpenFile
+	osStat     = os.Stat
+
+	apparmorUpdateHomedirsTunable = apparmor.UpdateHomedirsTunable
+	apparmorReloadAllSnapProfiles = apparmor.ReloadAllSnapProfiles
+)
+
+var (
+	// No path located under any of these top-level directories can be used.
+	// This is not a security measure (the configuration can only be changed by
+	// the system administrator anyways) but rather a check around unintended
+	// mistakes.
+	invalidPrefixes = []string{
+		"/bin/",
+		"/boot/",
+		"/dev/",
+		"/etc/",
+		"/lib/",
+		"/proc/",
+		"/root/",
+		"/sbin/",
+		"/sys/",
+		"/tmp/",
+	}
+
+	// TODO: remove this once we mount the root FS of a snap as tmpfs and
+	// become able to create any mount points. But for the time being, let's
+	// allow only those locations that can be supported without creating a
+	// mimic of "/".
+	validPrefixes = []string{
+		"/home/",
+	}
+)
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations["core.homedirs"] = true
+}
+
+func updateHomedirsConfig(config string) error {
+	snapStateDir := dirs.SnapdStateDir(dirs.GlobalRootDir)
+	if err := os.MkdirAll(snapStateDir, 0755); err != nil {
+		return err
+	}
+
+	configPath := filepath.Join(snapStateDir, "system-params")
+	f, err := osOpenFile(configPath, os.O_WRONLY|os.O_CREATE, 0644)
```

Comment:
Could this be `ioutil.WriteFile()` instead?

Reply:
Oh,I forgot about that function! Thanks for the suggestion, updated.